components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/internal/services/TemplateManagerService.java

```diff
@@ -55,24 +58,38 @@ public TemplateManagerService() {
     }
 
     public void createBusinessRuleFromTemplate(BusinessRuleFromTemplate businessRuleFromTemplate) {
-        // todo: verify next lower level
-        Map<String, Template> derivedTemplates = deriveTemplates(businessRuleFromTemplate);
+        // todo: CHECK THIS METHOD
+        // Derive templates from the given business rule from template
+        Map<String, Template> derivedTemplates = null;
+        try {
+            derivedTemplates = deriveTemplates(businessRuleFromTemplate);
+        } catch (TemplateManagerException e) {
+            log.error("Error in deriving templates", e);
+        }
         String businessRuleUUID = businessRuleFromTemplate.getUuid();
         try {
-            saveBusinessRuleDefinition(businessRuleUUID, businessRuleFromTemplate);
-            // Deploy templates, only if saving Business Rule definition is successful
-            // todo: (Q) is this ok?
+            saveBusinessRuleDefinition(businessRuleUUID, businessRuleFromTemplate); // todo : Implement method
+            // Deploy all derived templates, only if saving Business Rule definition is successful
             for (String templateUUID : derivedTemplates.keySet()) {
-                deployTemplate(templateUUID, derivedTemplates.get(templateUUID));
+                try {
+                    deployTemplate(templateUUID, derivedTemplates.get(templateUUID));
+                } catch (TemplateManagerException e) {
+                    log.error(e.getMessage()); // Exception is thrown from the above method itself
```

Comment:
Usually, we use log.error(e.getMessage(), e). So it prints the stack trace as well